model/lifecycle.go

```diff
@@ -442,16 +453,30 @@ func RestartVersions(ctx context.Context, versionsToRestart []*VersionToRestart,
 // RestartBuild restarts completed tasks associated with a given buildId.
 // If abortInProgress is true, it also sets the abort flag on any in-progress tasks.
 func RestartBuild(ctx context.Context, build *build.Build, taskIds []string, abortInProgress bool, caller string) error {
+	// Fetch tasks to reset before aborting to ensure we're including any needed display tasks.
+	tasks, err := getTasksWithDisplayTasks(taskIds)
```

Comment:
It seems this [logic](https://github.com/evergreen-ci/evergreen/pull/7057/files#diff-f50207584681d142f6e119549a8e2e22249b45dcbe3cb6f7570eb906be64299dR458-R469)  is duplicated across `RestartBuild` and `RestartVersion`, can we absorb it into `getTasksWithDisplayTasks`?

Reply:
I thought about that but it's kind of messy and I don't think actually simplifies it.

I'm actually going to approach this change differently so I'll be reverting these functions to closer to what they were.